src/components/Form/FormRatio.react.js

```diff
@@ -0,0 +1,81 @@
+// @flow
+
+import * as React from "react";
+import cn from "classnames";
+import { Grid } from "../";
+
+type Props = {|
+  +className?: string,
+  +step?: number,
+  +min?: number,
+  +max?: number,
+  +defaultValue?: number,
+  +value?: number,
+  +onChange?: (SyntheticInputEvent<EventTarget>) => mixed,
+|};
+
+type State = {|
+  internalValue: number,
+  hasChanged: boolean,
+|};
+
+class FormRatio extends React.PureComponent<Props, State> {
+  constructor(props: Props) {
```

Comment:
You can actually replace this whole `constructor(...)` block with the following:

```
state = {
    internalValue: this.props.defaultValue || this.props.value || 0,
    hasChanged: false,
};
```